preprocessor/src/main.rs

```diff
@@ -0,0 +1,279 @@
+use diesel::PgConnection;
+use dotenvy::dotenv;
+use feature_database::{imagedb, keypointdb, imagedb::ImageDatabase, keypointdb::KeypointDatabase, models};
+use feature_extraction::{akaze_keypoint_descriptor_extraction_def, DbKeypoints};
+use geotiff_lib::image_extractor;
+use geotiff_lib::image_extractor::{Datasets, MosaicDataset, MosaicedDataset};
+use homographier::homographier::raster_to_mat;
+use indicatif::{MultiProgress, ProgressBar};
+use tempfile::tempdir;
+
+use level_of_detail::calculate_amount_of_levels;
+use rayon as raycon;
+use raycon::Scope;
+
+use clap::Parser;
+use std::sync::{Arc, Mutex};
+
+pub mod level_of_detail;
+
+#[derive(Parser, Debug)]
+#[command(version, about, long_about = None)]
+struct Args {
+    /// The path to the folder where datasets are stored on disk
+    #[arg(short, long)]
+    dataset_path: Option<String>,
+
+    /// The path to a precomputed mosaic
+    #[arg(short, long)]
+    mosaic_path: Option<String>,
```

Comment:
hvis du bruger clap's subcommands, så kan du få den til at garantere at enten `dataset_path` eller `mosaic_path` er indstillet. se evt min branch, der er et eksempel på hvordan man kunne gøre noget lignende

Reply:
Noice. Will take a look at it then.